src/lib/transaction_logic/test/account_timing.ml

```diff
@@ -0,0 +1,194 @@
+open Core_kernel
+open Mina_transaction_logic
+open Currency
+open Mina_base
+open Mina_numbers
+
+type timing = (Global_slot.t, Balance.t, Amount.t) Account_timing.tt
+[@@deriving compare, sexp]
+
+type txn_result = (timing, Error.t) Result.t [@@deriving compare, sexp]
+
+let final_vesting_slot ~initial_minimum_balance ~cliff_time ~cliff_amount
+    ~vesting_period ~vesting_increment =
+  let open Unsigned in
+  let to_vest =
+    Balance.(initial_minimum_balance - cliff_amount)
+    |> Option.value_map ~default:UInt64.zero ~f:Balance.to_uint64
+  in
+  let incr = Amount.to_uint64 vesting_increment in
+  let periods =
+    let open UInt64 in
+    if compare (rem to_vest incr) zero = 0 then div to_vest incr
```

Comment:
`div to_vest incr` is computed in both arms of the `if`, so it could be factored out

Reply:
I factored the division out in 6c496a4d9a. Not really sure, which version I like better, though.